src/owlwatch/owlwatch.py

```diff
@@ -1,477 +1,525 @@
+            print(first + " Vs. " + second)
+            print("Comparison result: " + result_format + status + Style.RESET_ALL)
+            print("Matches:", len(correct))
+            print("Not found in " + second + ": ", len(missing))
+            print("Type mismatches: ", len(distinct))
+            if msg != "":
+                print("Details: \n" + msg + '\n')
+    else:
+        logging.info("Didn't find anything to do...")
+    logging.info("This is the end.")
+
+def check_empty_panels_subparser(subparsers):
+    """Adds a subparser for check_empty_panels command"""
+    help_txt = """Check whether a certain panel is empty or not"""
+    parser_check_empty_panel = subparsers.add_parser(CHK_PANEL,
+                                       help=help_txt)
+
+    parser_exc_group = parser_check_empty_panel.add_mutually_exclusive_group(required=True)
+    parser_exc_group.add_argument('--projects', dest='projects',
+                                  help='e.g.: "project1, project2"')
+    parser_exc_group.add_argument('--check_all', dest='check_all', action='store_true',
+                                  help='Check the selected panels for all ES instances on the current host')
+
+    parser_check_empty_panel.add_argument('--port', dest='port',
+                                          default=9200, type=int,
+                                          help='e.g.: 9200')
+    parser_check_empty_panel.add_argument('--panels', dest='panels',
+                                          required=True,
+                                          help='e.g.: "panel1, panel2"')
+    parser_check_empty_panel.add_argument('-u', dest='user',
+                                          help='e.g.: myuser')
+    parser_check_empty_panel.add_argument('-p', dest='password',
+                                          help='e.g.: mypassword')
+    parser_check_empty_panel.add_argument('-s', dest='storage_dir',
+                                          default='/tmp/',
+                                          help='e.g.: /path_where_ES_can_be_found_on_the_host')
+
+
+def main():
+    """This is the Owlcave, where everything starts"""
+    args = parse_args()
+
+    configure_logging(args.info, args.debug)
+
+    logging.info("** The Owl is watching **")
+
+    if args.subparser_name == CHK_PANEL:
+        check_empty_panels = CheckEmptyPanels()
+        projects = {}
+        if args.check_all:
+            storage_dir =args.storage_dir
+            projects = check_empty_panels.get_containers(storage_dir)
+        else:
+            for project in args.projects.split(', '):
+                projects[project] = args.port
+
+        for panel in args.panels.split(', '):
+            for project in projects:
+                check_empty_panels.check_panel(project, projects[project], panel,
+                                               check_all=args.check_all,
+                                               user=args.user, password=args.password)

```

Comment:
Instead of printing result inside `check_panel`, please return a data structure (e.g.: https://github.com/chaoss/grimoirelab-sigils/blob/master/src/owlwatch/schema/model.py#L123) and print results here. The reason to do this is because the method you are calling should be a library method, while `owlwatch.py` is acting as front-end for that library here.